metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,303 @@
+import click
+from metaflow import namespace
+from metaflow.util import resolve_identity
+from metaflow.exception import CommandException, MetaflowNotFound
+from metaflow.exception import MetaflowNamespaceMismatch
+
+# a rough emulation of itertools.zip_longest for our needs
+# because it's not available in py2
+def zip_longest(list1, list2):
+    if (len(list1) < len(list2)):
+        list1.extend([None] * (len(list2) - len(list1)))
+    elif (len(list1) > len(list2)):
+        list2.extend([None] * (len(list1) - len(list2)))
+    return zip(list1, list2)
+
+def format_tags(system_tags, all_tags):
+    import sys
+
+    max_length = max([len(t) for t in all_tags] if all_tags else [0])
+
+    all_tags = sorted(all_tags)
+    if sys.version_info[0] < 3:
+        all_tags = [t if t in system_tags else '*%s*' % t.encode('utf-8') for t in all_tags]
+    else:
+        all_tags = [t if t in system_tags else '*%s*' % t for t in all_tags]
+
+    num_tags = len(all_tags)
+
+    # We consider 4 spaces in between column and we consider 120 characters total
+    # width and we want 120 >= column_count*max_length + (column_count - 1)*4
+    column_count = 124 // (max_length + 4)
+    if column_count == 0:
+        # Make sure we have at least 1 column even for very very long tags
+        column_count = 1
+        words_per_column = num_tags
+    else:
+        words_per_column = (num_tags + column_count - 1) // column_count
+
+    # Extend all_tags by empty tags to be able to easily fill up the lines
+    addl_tags = words_per_column * column_count - num_tags
+    if addl_tags > 0:
+        all_tags.extend([' ' for _ in range(addl_tags)])
+
+    lines = ["",
+             "System tags and *user tags*; only *user tags* can be removed",
+             "--------------------------------------------------------",
+             ""]
+
+    for i in range(words_per_column):
```

Comment:
would it be controversial to suggest a three-column format

`[run-id] [user|sys ] [tag]`

this would be much more CLI-friendly: compatible with `cut`, `grep` etc.

Reply:
I think the idea was to use the client if you wanted a parseable thing and use this as a "pretty printing". But I don't mind either way as long as the info is clear. Note that there can be a *lot* of tags so a single column format as you are suggesting would make for a lot of text particularly if you dedup tags based on run ids (which is not currently the case).